core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -520,6 +520,8 @@ object SparkSubmit {
       }
       if (args.principal != null) {
         require(args.keytab != null, "Keytab must be specified when the keytab is specified")
+        sysProps.put("spark.yarn.keytab", args.keytab)
+        sysProps.put("spark.yarn.principal", args.principal)
```

Comment:
@harishreedharan I see you changed this part of code last time. If we want to pass these two arguments to Spark SQL, what is the recommended way?